libs/prelude/Prelude/Algebra.idr

```diff
@@ -286,44 +160,19 @@ class (VerifiedBoundedJoinSemilattice a, VerifiedBoundedMeetSemilattice a, Verif
 ||| + Unity for `<.>`:
 |||     forall a,     a <.> unity     == a
 |||     forall a,     unity <.> a     == a
-||| + InverseM of `<.>`:
-|||     forall a,     a <.> inverseM a == unity
-|||     forall a,     inverseM a <.> a == unity
+||| + InverseM of `<.>`, except for neutral
+|||     forall a /= neutral,  a <.> inverseM a == unity
+|||     forall a /= neutral,  inverseM a <.> a == unity
 ||| + Distributivity of `<.>` and `<->`:
 |||     forall a b c, a <.> (b <+> c) == (a <.> b) <+> (a <.> c)
 |||     forall a b c, (a <+> b) <.> c == (a <.> c) <+> (b <.> c)
-class RingWithUnity a => Field a where
-  inverseM : a -> a
+class (RingWithUnity a, Eq a) => Field a where
+  inverseM : (x : a) -> (notId : x == neutral = False) -> a
```

Comment:
I agree that two equalities `==` and `=` can be of different complexity to proof them or their negation. The `Eq` class seems more being a run-time check (even it is not so). 

Perhaps we should try to write a pair functions to establish the equivalence of `x == something = False` and `Not (x = something)`. Or rather of  `x == something = True` and `x = something`. Perhaps these functions should be members of a new `VerifiedEq` class. Then we will be able to provide a propositional inequality `Not (x = neutral)` via a call of such function (in instance declaration) while the `Field` class still being free of dependency on `Eq`.